### PR TITLE
refactor: reconstruct the judgment of OwningGatewayLabels

### PR DESCRIPTION
**Pull request by ShyunnY**
**What type of PR is this?**
* refactor: reconstruct the judgment of OwningGatewayLabels

**What this PR does / why we need it**:
I refactored the judgment of OwningGatewayLabels in the `resource_provider.go` file.

First of all, I think this logic can be abstracted into a function without redundant repeated writing.

Secondly, make it backward compatible.

### CODE REVIEW

**Review by zirain — DISMISSED** [review]

**Review by shawnh2 — APPROVED** [review]

File: internal/infrastructure/kubernetes/proxy/resource_provider.go [hunk 1]
```diff
@@ -294,3 +294,10 @@ func (r *ResourceRender) HorizontalPodAutoscaler() (*autoscalingv2.HorizontalPod
 
 	return hpa, nil
 }
+
+// lackOwningGatewayLabels Check if labels are missing some OwningGatewayLabels
+func lackOwningGatewayLabels(labels map[string]string) bool {
```
Comment (arkodg) [inline internal/infrastructure/kubernetes/proxy/resource_provider.go:299]:
```suggestion
func OwningGatewayLabelsAbsent(labels map[string]string) bool {
```

Comment (ShyunnY) [reply internal/infrastructure/kubernetes/proxy/resource_provider.go]: This does indeed seem more suitable, I have made the changes.

File: internal/infrastructure/kubernetes/proxy/resource_provider_test.go [hunk 2]
```diff
@@ -746,3 +746,55 @@ func loadHPA(caseName string) (*autoscalingv2.HorizontalPodAutoscaler, error) {
 	_ = yaml.Unmarshal(hpaYAML, hpa)
 	return hpa, nil
 }
+
+func TestLackOwnerGatewayLabels(t *testing.T) {
```
Comment (cnvergence) [inline internal/infrastructure/kubernetes/proxy/resource_provider_test.go:750]:
```suggestion
func TestOwningGatewayLabelsAbsent((t *testing.T) {
```
Please also change it here

Comment (ShyunnY) [reply internal/infrastructure/kubernetes/proxy/resource_provider_test.go]: This was an oversight on my part, I have corrected it

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Pull request merged**